### ISSUE TITLE
`process` is a very bad function name

### ISSUE THREAD

**Issue by s-panferov** [mobxjs/mobx-state-tree#399]
Hello. Recently I faced with a bug in my code that I can't make ReactHotLoader work because it uses `process.env.NODE_ENV` global variable to check an environment. But if you have `import { process } from 'mobx-state-tree'` require, webpack fails to inline `process.env.NODE_ENV` variable.

This is very bad because a lot of people may face it and it's really hard to understand what is going on.

**Comment by tinchoz49** [comment]
I think the same as you. What I'm doing to solve the problem is using 
`import { process as co } from 'mobx-state-tree'`

**Comment by aretecode** [comment]
`fromGeneratorToAsync`

**Comment by nickbreaton** [comment]
What if it was renamed to `fromGeneratorToAsync` as suggested by @aretecode, but it was implied by using a generator function in actions.
```js
types
  .model({
    name: 'default'
  })
  .actions(self => ({
    changeName(name) {
      self.name = name
    },
    *fetchRandomName() { 
      const response = yield fetch('.....')
      self.name = response.data
    }
  }))
```

instead of

```js
  types
    .model({
      name: 'default'
    })
    .actions(self => ({
      changeName(name) {
        self.name = name
      },
      /* this would replace 'process', and would still be valid, but verbose */
      fetchRandomName: fromGeneratorToAsync(function* () { 
        const response = yield fetch('.....')
        self.name = response.data
      })
    }))
```

This can be detected by checking the constructor name:

```js
(function a() {}).constructor.name  // => "Function"
(function* b() {}).constructor.name  // => "GeneratorFunction"
```

**Comment by mattiamanzati** [comment]
@nickbreaton In the past it worked like that, but was changed due to problems imlied by transpilation. See #279.

**Comment by nickbreaton** [comment]
@mattiamanzati Sorry about that I checked Babel, but not TS. 

I'm just spit balling, but what about something like this?
```js
types
.model({ name: '' })
.actions(self => ({
  changeName(name) {
    self.name = name
  }
}))
.processes(self => ({
  *fetchRandomName() {
    // ...
  }
}))
```

It would allow people to avoid importing `process`.

**Comment by mattiamanzati** [comment]
Could potentially be a solution, I would like more something like

.compose(processes(self =>({
....
})

so that eventually other community-provided model enhancer could be provided.

**Comment by nickbreaton** [comment]
Personally I would be in favor of adding both `.processes` and `.compose`. One shortcoming of Redux, IMO, is that there is no baked in solution to async actions. I think `.processes` would show the "blessed" way of doing async actions out of the box, and `.compose` would be there for anyone else wanting to provide a model enhancer.

**Comment by tinchoz49** [comment]
in my case what i'm using is a map function to process: 
```javascript
const mapToProcess = (props = {}) => {
  return Object.keys(props).reduce((result, key) => {
    if (
      props[key].constructor &&
      props[key].constructor.name === 'GeneratorFunction'
    ) {
      result[key] = co(props[key]);
      return result;
    }

    result[key] = props[key];
    return result;
  }, {});
};
```

so in your actions definitios you can do this: 

```javascript
...
.actions(self => mapToProcess({
  changeName(name) {
    ...
  },
  *fetchRandomName() {
    ...
  }
}))
```

**Comment by mweststrate** [comment]
The problem of some of the above proposed solutions is that they don't support non-exposed processes (processed that are only available in the closure). I think a nice alias for `process` should solve that. E.g.:

* `co`
* `routine`
* `story`
* `flow`
* `saga` (lol)

Personally I like `flow` best

**Comment by mattiamanzati** [comment]
Flow is a very good one!

**Comment by mweststrate** [comment]
any tools out there that secretly inject flow globals :-P ?

Op ma 25 sep. 2017 om 12:00 schreef Mattia Manzati <notifications@github.com

**Comment by nickbreaton** [comment]
I personally think that having both `process` and `flow` (or whatever name is decided on) could be confusing to users. Also, a `flow` would then have [action types that are prefixed with `process_`](https://github.com/mobxjs/mobx-state-tree/blob/9568fd3/src/core/action.ts#L4-L9). 

I would be in favor of renaming `process` to `flow`, and all subsequent variable / type names.

Either way, I would love to help contribute to such an awesome project!

**Comment by d4rky-pl** [comment]
how about [asyncAction](https://github.com/mobxjs/mobx-utils#asyncaction)?

**Comment by tinchoz49** [comment]
I like the abstraction behind the word: `activity` but i can live with `flow` too :smile:

**Comment by nickbreaton** [comment]
@mattiamanzati This can probably be closed as #435 is merged.
